### PR TITLE
dx12/vk/ogl: index_base_offset implementation

### PR DESCRIPTION
**Pull request by jarveson**
Implements use of 'vertex_data_base_index' register, taking into account the base index when creating/using index buffers
notably fixing mickey in castle of illusion

Also missing library fix for debug building

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ namespace
 				rsx::method_registers.current_draw_clause.primitive,
 				rsx::method_registers.restart_index_enabled(),
 				rsx::method_registers.restart_index(), command.ranges_to_fetch_in_index_buffer,
-				[](auto prim) { return !vk::is_primitive_native(prim); });
+				[](auto prim) { return !vk::is_primitive_native(prim); }, rsx::method_registers.vertex_data_base_index());
```
Comment (kd-11) [inline rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp]: Not a huge fan of predicates dangling in the middle of function lists, to allow expansion onto more than one line without messing up the flow. Move the vertex base index argument before the predicate function. Other than that it should be good to go.

**Comment by SakataGintokiYT** [comment]
**Before/After Comparisons (Click to Expand)**

<details>
 <summary>LEGO Star Wars III: The Clone Wars</summary>

before
after
</details>

<details>
 <summary>Castle of Illusion Starring Mickey Mouse</summary>

before
after
</details>

<details>
 <summary>RIO</summary>

before
after
</details>

<details>
 <summary>Valkyria Chronicles</summary>

before
after
</details>

**Comment by miguel-nunes** [comment]
#3916

**Comment by JohnGodgames** [comment]
## Fixed Knights Contract

**Before:**
kc before

**After:**
kc after

**Comment by Xcedf** [comment]
Another fix Yakuza 3 artifacts
Before:

After:

**Comment by Kravickas** [comment]
**PR**
pr

**Masterino**
master

**Comment by kamer1337** [comment]
Blood Drive

Before:
before
After:
after

**Comment by miguel-nunes** [comment]
Folks, some details with Valkyria Chronicles, that maybe it's still related to this PR:

The river in the first skirmish is still being rendered with wrong angles. Sometimes there're errors in the game filling the river, water being rendered in wrong places.

All tests I did in Linux, so if someone confirms or denies in Windows, I'll collect the proper logs and images to feed this ticket.

Cheers,

**Comment by SakataGintokiYT** [comment]
@miguel-nunes 
This game needs strict rendering mode.

**Comment by Asinin3** [comment]
Before/After Comparisons (Click to Expand)
<details>
 <summary>LEGO Batman 3</summary>

before
after
</details><details>
 <summary>LEGO Jurassic World</summary>

before
after
</details>

**Pull request merged**